presto-geospatial/src/test/java/com/facebook/presto/geospatial/TestGeoQueries.java

```diff
@@ -38,6 +42,18 @@ protected void registerFunctions()
         functionAssertions.getMetadata().addFunctions(extractFunctions(plugin.getFunctions()));
     }
 
+    @Test
+    public void testGeometryGetObjectValue()
+            throws Exception
+    {
+        Slice slice = GeoFunctions.stPoint(1.2, 3.4);
+        int[] offsets = {0, slice.length()};
+        boolean[] valueIsNull = {false};
+        VariableWidthBlock block = new VariableWidthBlock(1, slice, offsets, valueIsNull);
```

Comment:
Creating a block manually is generally a bad idea, even in a test. The `Type` should be used to create the block:
```java
BlockBuilder builder = GEOMETRY.createBlockBuilder(new BlockBuilderStatus(), 1);
GEOMETRY.writeSlice(builder, slice);
Block block = block.build();

Reply:
The contract between blocks and types is that the type owns the block and representation of the data, so only the type should read or write it.